src/clients/TorrentClient.ts

```diff
@@ -10,6 +10,7 @@ import Deluge from "./Deluge.js";
 let activeClient: TorrentClient;
 
 export interface TorrentClient {
+	getDownloadDir?: (searchee: Searchee) => Promise<string>;
```

Comment:
Because `getDownloadDir` is not present on the rTorrent integration, the optional (?) was necessary. Once it's on all clients it can be removed (it's in my task's notes)....

So you would want something like 	

```suggestion
	getDownloadDir?: (
		searchee: Searchee
	) => Promise<Result<string, InjectionResult.FAILURE>>;
```

I'm not really familiar with the Result, and it only appears to be used in rTorrent and Transmission - I didn't use it in deluge and I don't see it in qBit either...and those are the main places I've been working clientwise.

We can expand its usage though, either just in these functions or more broadly if you desire.

Reply:
I would use a separate type than InjectionResult.FAILURE, but yes. I can try making that refactor as I'll be playing with the rTorrent implementation